Document how to use daskhub unreleased version of the helm chart (#10)
2i2c-org/farallon-image: README.md

   ```bash
      hubploy deploy <hub-name> hub prod
   ```

## How to use an unreleased helm chart

Sometimes you might need to use the latest version of a helm chart and that isn't yet released.
The following steps shows how to do so for the DaskHub project, but the pattern
should be replicable for others.

1. Clone the repo holding the latest version of the helm-chart and checkout the wanted branch:

   *DaskHub example:*

   ```bash
   git clone https://github.com/dask/helm-chart.git

>    git checkout master

Reviewer: ````suggestion
   git checkout master
   ```

````

I think we messed up somehow related to the use of three tickmarks interfere with the use of three tickmarks in the suggestion. In this commen't suggestion I use four tickmarks to be able to use three from within.

Author: Hehe :sweat_smile: Interesting

   ```bash
      hubploy deploy <hub-name> hub prod
   ```

## How to use an unreleased helm chart

Sometimes you might need to use the latest version of a helm chart and that isn't yet released.
The following steps shows how to do so for the DaskHub project, but the pattern
should be replicable for others.

1. Clone the repo holding the latest version of the helm-chart and checkout the wanted branch:

   *DaskHub example:*

>    ```bash
>       git clone https://github.com/dask/helm-chart.git
>       git checkout master
>    ```

Reviewer: ```suggestion
   ```bash
   git clone https://github.com/dask/helm-chart.git
   git checkout master
   ```
```

> 2. Inside the `yaml` file holding the dependencies, add the local path to the helm-chart of the project
> you just checked out and use the version inside the `chart.yaml` or run `chartpress` to generate one if
> that is not available:

Reviewer: ```suggestion
2. Reference the version of the unpackaged Helm chart (of the locally cloned git repository) in the dependencies `yaml` file of your project (either `requirements.yaml` or `Chart.yaml`). This version should be available in the`Chart.yaml` file (of the cloned repository). If no version is available it may be because it is using `chartpress` to set it, so you need to run `chartpress` in that repository first to generate it.
```

I wanted to iterate on this sentence a bit, perhaps you can help adjust my attempt to something that makes things clear?

Author: I edited your comment, hope that's ok. What do you think about it? :D

Reviewer: Nice! I did another iteration again on the last sentence, feel free to edit directly or do as you please!

```diff
-If no version is available it may be because it is using `chartpress` to set it, so you need to run `chartpress` in that repository first to generate it.
+If the version is set to be a placeholder or outdated, it may be because it is maintained using `chartpress`, then run `chartpress` in that repository first to update it.
```

```suggestion
2. Reference the version of the unpackaged Helm chart (of the locally cloned git repository) in the dependencies `yaml` file of your project (either `requirements.yaml` or `Chart.yaml`). This version should be available in the`Chart.yaml` file (of the cloned repository). If the version is set to be a placeholder or outdated, it may be because it is maintained using `chartpress`, then run `chartpress` in that repository first to update it.
```

Author: No, that's perfect :heart: Thanks Erik, I really like how it turned out.

Reviewer: Thanks for iterating on it with me and working on this documentation @GeorgianaElena! :sunflower:

   *DaskHub example:*

>    ```bash
>       dependencies:
>        - name: daskhub
>          repository: file://<path-to-your-daskhub-checkout>/daskhub
>          version: <version-from-chart.yaml-of-daskhub>
>    ```

Reviewer: ````suggestion
   ```yaml
   dependencies:
     - name: daskhub
       repository: file://<path-to-your-daskhub-checkout>/daskhub
       version: <version-from-chart.yaml-of-daskhub>
   ```
````

3. Run an update any time the dependencies (in `requirements.yaml` or `Chart.yaml`) of the chart you
just cloned changes.

      *DaskHub example:*

>       ```bash
>          # from inside the dask/helm-chart checkout
>          helm dep up daskhub
>       ```

Reviewer: ```suggestion
      ```bash
      # from inside the dask/helm-chart checkout
      helm dep up daskhub
      ```
```

   ```bash
      hubploy deploy <hub-name> hub prod
   ```

## How to use an unreleased helm chart

Sometimes you might need to use the latest version of a helm chart and that isn't yet released.
The following steps shows how to do so for the DaskHub project, but the pattern
should be replicable for others.

1. Clone the repo holding the latest version of the helm-chart and checkout the wanted branch:

   *DaskHub example:*

   ```bash
      git clone https://github.com/dask/helm-chart.git
      git checkout master
   ```
2. Inside the `yaml` file holding the dependencies, add the local path to the helm-chart of the project
you just checked out and use the version inside the `chart.yaml` or run `chartpress` to generate one if
that is not available:

   *DaskHub example:*

   ```bash
      dependencies:
       - name: daskhub
         repository: file://<path-to-your-daskhub-checkout>/daskhub
         version: <version-from-chart.yaml-of-daskhub>
   ```

3. Deploy the changes.

>    **Note** If after deploying these changes some of the services don't show up and they are part of projects that

Reviewer 2: I think you should run `helm dep up` regardless of what happens - I think otherwise this *will* happen.

Reviewer 2: Or rather, this should be run any time the dependencies (in requirements.yaml or Chart.yaml) of the chart you just cloned changes.

Reviewer: :+1: 

- `helm repo update`: fetching the latest information from [a Helm chart repository's index.yaml](https://jupyterhub.github.io/helm-chart/index.yaml) file which lists all available versions and charts in the Helm chart repository. Not having done this can make a `helm upgrade jupyterhub/jupyterhub --version 1.2.3` fail because one doesn't have information about the new published version locally yet even though it is available.
- `helm dep update`: this will use available chart information to update a local copy put in the /charts folder within the Helm chart, it is a folder where local copies of Helm chart reside.

If you use a Helm chart dependency being `file://`, then you won't need `helm repo update` for that dependency at least, but you always need `helm dep up` I think to copy the dependency to the `/charts` folder alongside `/templates`

Author: Thank you for all the useful info :heart:
